gitlab/protocols/__init__.py

```diff
@@ -0,0 +1,3 @@
+"""
+Defines protocols
+"""
```

Comment:
Same as above: rather than having a separate filepath for this I think this belongs along with the backends in `gitlab/_backends`. Probably both in something like `gitlab/_backends/base.py` to keep it simple for now, also simplifies the imports :)